typescript/schedules.md

```diff
@@ -0,0 +1,1255 @@
+   */
+  trigger(): Promise<void>;
+
+  /**
+   * Run though the specified time period(s) and take Actions as if that time passed by right now, all at once. The
+   * Overlap Policy can be overridden for the scope of the Backfill.
+   */
+  backfill(options: Backfill | Backfill[]): Promise<void>;
+
+  /**
+   * Pause the Schedule
+   */
+  pause(note?: string): Promise<void>;
+
+  /**
+   * Unpause the Schedule
+   */
+  unpause(note?: string): Promise<void>;
+
+  /**
+   * Fetch the Schedule's description from the Server
+   */
+  describe(): Promise<ScheduleDescription>;
+
+  /**
+   * Get a handle to the most recent Action started
+   */
+  lastAction<T extends ScheduleActionType>(): Promise<HandleFor<T>>;
+
+  /**
+   * Readonly accessor to the underlying ScheduleClient
+   */
+  readonly client: ScheduleClient;
+}
+
+export type Base64 = string;
+
+export interface Schedule {
+  /**
+   * Schedule Id
+   *
+   * We recommend using a meaningful business identifier.
+   */
+  id: string;
+
+  /**
+   * Additional non-indexed information attached to the Schedule. The values can be anything that is
+   * serializable by the {@link DataConverter}.
+   */
+  memo?: Record<string, any>;
+
+  /**
+   * Additional indexed information attached to the Schedule. More info:
+   * https://docs.temporal.io/docs/typescript/search-attributes
+   *
+   * Values are always converted using {@link JsonPayloadConverter}, even when a custom Data Converter is provided.
+   */
+  searchAttributes: SearchAttributes;
+
+  // TODO flatten this?
+  info: ScheduleInfo;
+}
+
+/**
+ * The current Schedule details. They may not match the Schedule as created because:
+ * - some fields in the state are modified automatically
+ * - the schedule may have been modified by {@link ScheduleHandle.update} or
+ *   {@link ScheduleHandle.pause}/{@link ScheduleHandle.unpause}
+ */
+export interface ScheduleDescription {
+  /** When Actions should be taken */
+  spec: RequireAtLeastOne<ScheduleSpec, 'calendars' | 'intervals'>;
+
+  /**
+   * Which Action to take
+   */
+  action: ScheduleActionOptions;
+
+  /**
+   * Controls what happens when an Action would be started by a Schedule at the same time that an older Action is still
+   * running.
+   *
+   * @default {@link ScheduleOverlapPolicy.Skip}
+   */
+  overlap: ScheduleOverlapPolicy;
+
+  /**
+   * The Temporal Server might be down or unavailable at the time when a Schedule should take an Action. When the Server
+   * comes back up, `catchupWindow` controls which missed Actions should be taken at that point. The default is one
+   * minute, which means that the Schedule attempts to take any Actions that wouldn't be more than one minute late. It
+   * takes those Actions according to the {@link ScheduleOverlapPolicy}. An outage that lasts longer than the Catchup
+   * Window could lead to missed Actions. (But you can always {@link ScheduleHandle.backfill}.)
+   *
+   * @default 1 minute
+   */
+  catchupWindow: Milliseconds;
```

Comment:
It sure is confusing that if I have in my code `catchupWindow: 1000` and `at: 1000` the first one means 1000ms and the second one means 1000s. Can't you just accept the same time interval for everything and if something can't accept a granularity lower than seconds either trunc or error? This has gone a bit type crazy and users won't be able to tell when reading. We should use the same durations throughout our API.

Reply:
Changing from this:

<img width="407" alt="image" src="https://user-images.githubusercontent.com/251288/184174342-34235b42-1508-41fd-8534-2d815195c547.png">

to this?

<img width="560" alt="image" src="https://user-images.githubusercontent.com/251288/184174645-09efc26c-2032-46ba-90de-bd5232c4c4a1.png">

Reply:
How do you accept durations for retry policy? Or start workflow timeouts? Can we use the same duration form everywhere?

Reply:
Those are all `string | number` milliseconds, which is the def of `Milliseconds`. I'd like to change those to `Milliseconds` too.

Reply:
I'd be fine with keeping `string | number` and clarify in the docstring of each option in the SDK that we accept ms formatted string or milliseconds.
Don't mind defining a common `Milliseconds` type either though.

Reply:
Nevermind re: `Milliseconds`. While I find it easier to read the types of, it's easier to know what to do when reading the in-editor doc if it's the other way:

<img width="1559" alt="image" src="https://user-images.githubusercontent.com/251288/184241116-69e2bd2a-7d38-4776-ba12-993c9b1ddf8c.png">